lib/task_listener.js

```diff
@@ -322,43 +322,40 @@ export default class TaskListener extends EventEmitter {
    * directly from the TC queue service, then calling runTask with the
    * resulting task set.
    */
-  async coalesceClaim(claim) {
+  async applySuperseding(claim) {
     let task = claim.task;
     let taskId = claim.status.taskId;
 
     try {
-      // if the task is not coalescible, then the taskset is just the one claim
-      if (!task.payload.coalescer) {
-        this.runtime.log('not coalescing', {taskId, message: 'no coalescer in payload'});
+      // if the task is not set up to supersede anything, then the taskset is just the one claim
+      if (!task.payload.supersederUrl) {
+        this.runtime.log('not superseding', {taskId, message: 'no supersederUrl in payload'});
         return [claim];
       }
 
-      let routes = task.routes;
-      let routePrefix = task.payload.coalescer.routePrefix + ".";
-      routes = routes.filter(r => r.startsWith(routePrefix));
-      if  (routes.length != 1) {
-        this.runtime.log('not coalescing', {taskId, message: 'no single coalescer route in payload'});
+      let supersederUrl = task.payload.supersederUrl;
+
+      let tasks = (await this.fetchSupersedingTasks(supersederUrl, taskId));
+      if (!tasks || tasks.length == 0 || tasks[0] === taskId) {
```

Comment:
Actually the check shouldn't be there at all.  I put it in when I was thinking we would only claim until we hit the initial taskId -- then I realized that was wrong, and missed changing this bit.  You're a very thorough reviewer!